### ISSUE TITLE
Android: Failed to build standalone app

### ISSUE THREAD

**Issue by TRayS2** [expo/turtle#64]
I tried to create a gitlab-ci that builds the apk automatically. But when I get to the final step of actually building with turtle-cli, it throws an error:
```javascript
Failed to build standalone app
  err: TypeError: Cannot read property 'android' of undefined
      at Object.getOrCreateCredentials [as default] (/usr/local/share/.config/yarn/global/node_modules/turtle-cli/src/builders/utils/android/credentials.ts:22:45)
      at buildAndroid (/usr/local/share/.config/yarn/global/node_modules/turtle-cli/src/builders/android.ts:22:51)
      at <anonymous>
  platform: "android"
```

I can't seem to fix it and I've tried running the turtle process manually on the docker image with no luck.
The dockerfile is use:
```dockerfile
FROM dsokal/expo-turtle-android:latest
RUN sudo yarn global add node-gyp bcrypt@3.0.2 gulp-cli
RUN sudo yarn global add turtle-cli
RUN sudo turtle setup:android
```
I've setup the environmental variables via the gitlab-runner page and the commands I run within the `.gitlab-ci.yml` are:
```
     - mkdir ./dist
    - git submodule sync --recursive
    - git submodule update --init --recursive
    - yarn install --prod
    - sudo turtle build:android -o ./dist/applet.apk -u $EXPO_USERNAME -p $EXPO_PASSWORD  #! DEPHRECATED: > ./dist/stdout.txt 2> ./dist/stderr.txt
```
I tried to look into `/node_modules/turtle-cli/src/builders/utils/android/credentials.ts:22:45`, but it doesn't seem to exist (probably due to src code). So now I'm lost.
If anyone could help me, that would be amazing :)
Thank you in advance

**Comment by dsokal** [comment]
You don't have `android.package` key in your `app.json`. Please add that in and try running a build again.

_(side note: you don't need `sudo` neither for yarn nor for turtle)_

**Comment by TRayS2** [comment]
Thank you for such a quick response! although I definitely checked inside `app.json` (to be sure, I ran it twice too). But I have tried both `expo.android.package` && `android.package`. 
```json
{
  "ios": {
    "icon": "./src/assets/images/icon.png",
    "supportsTablet": true,
    "bundleIdentifier": "local.company.applet"
  },
  "android": {
    "icon": "./src/assets/images/icon_android.png",
    "package": "local.company.applet"
  }
```
```json
  "expo": {
    "ios": {
      "icon": "./src/assets/images/icon.png",
      "supportsTablet": true,
      "bundleIdentifier": "local.company.applet"
    },
    "android": {
      "icon": "./src/assets/images/icon_android.png",
      "package": "local.company.applet"
    },
  ...
}
```
I am sincerely lost.

**Comment by dsokal** [comment]
That's totally not possible. Have you committed the changes? It should be `expo.android.package`

**Comment by dacevedo12** [comment]
Im having the same error right now. It didn't happen with 0.7.1

**Comment by dsokal** [comment]
The fix will be included in the next release of turtle-cli.

@TRayS2 In the meantime, please see [@dacevedo12's comment](https://github.com/expo/turtle/issues/64#issuecomment-492397973) and [this PR's description](https://github.com/expo/turtle/pull/65).

**Comment by TRayS2** [comment]
@dsokal I thank you for your time and it seems your tips have helped.
I removed the sudo prefix between every command by appending the `yarn global bin` command output to my `PATH` env after globally installing the dependencies.
@dacevedo12 has made me look into my env of gitlab ci and indeed I was stupid enough to turn the  `KEYSTORE_PASSWORD` & `KEY_PASSWORD` around. So thank you for that too :)
Now I tried to set the `KEYSTORE_BASE64` env, but that did not seem to work. Als the `LOG_LEVEL` env did not work either, but the building was succesful. So you have my gratitude.
